src/KeyringController.ts

```diff
@@ -785,21 +842,120 @@ class KeyringController extends EventEmitter {
     }
     throw new Error(
-      `No keyring found for the requested account. Error info: ${errorInfo}`,
+      `${KeyringControllerError.NoKeyring}. Error info: ${errorInfo}`,
+    );
+  }
+
+  // =======================
+  // === Private Methods ===
+  // =======================
+
+  /**
+   * Update memStore Keyrings
+   *
+   * Updates the in-memory keyrings, without persisting.
+   */
+  async #updateMemStoreKeyrings(): Promise<Json> {
+    const keyrings = await Promise.all(
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      this.keyrings.map(this.#displayForKeyring),
+    );
+    return this.memStore.updateState({ keyrings });
+  }
+
+  /**
+   * Create First Key Tree.
+   *
+   * - Clears the existing vault.
+   * - Creates a new vault.
+   * - Creates a random new HD Keyring with 1 account.
+   * - Makes that account the selected account.
+   * - Faucets that account on testnet.
+   * - Puts the current seed words into the state tree.
+   *
+   * @returns A promise that resolves if the operation was successful.
+   */
+  async #createFirstKeyTree(): Promise<null> {
+    await this.#clearKeyrings();
+
+    const keyring = await this.addNewKeyring(KeyringType.HD);
+    if (!keyring) {
+      throw new Error(KeyringControllerError.NoKeyring);
+    }
+
+    const [firstAccount] = await keyring.getAccounts();
+    if (!firstAccount) {
+      throw new Error(KeyringControllerError.NoAccountOnKeychain);
+    }
+
+    const hexAccount = normalizeAddress(firstAccount);
+    this.emit('newVault', hexAccount);
+    return null;
+  }
+
+  /**
+   * Restore Keyring Helper
+   *
+   * Attempts to initialize a new keyring from the provided serialized payload.
+   * On success, returns the resulting keyring instance.
+   *
+   * @param serialized - The serialized keyring.
+   * @param serialized.type - Keyring type.
+   * @param serialized.data - Keyring data.
+   * @returns The deserialized keyring or undefined if the keyring type is unsupported.
+   */
+  async #restoreKeyring(
+    serialized: SerializedKeyring,
+  ): Promise<Keyring<Json> | undefined> {
+    const { type, data } = serialized;
+
+    let keyring: Keyring<Json> | undefined;
+    try {
+      keyring = await this.#newKeyring(type, data);
+    } catch {
+      // Error
+    }
+
+    if (!keyring) {
+      this.unsupportedKeyrings.push(serialized);
+      return undefined;
+    }
+
+    // getAccounts also validates the accounts for some keyrings
+    await keyring.getAccounts();
+    this.keyrings.push(keyring);
+    return keyring;
+  }
+
+  /**
+   * Get Keyring Class For Type
+   *
+   * Searches the current `keyringBuilders` array
+   * for a Keyring builder whose unique `type` property
+   * matches the provided `type`,
+   * returning it if it exists.
+   *
+   * @param type - The type whose class to get.
+   * @returns The class, if it exists.
+   */
+  #getKeyringBuilderForType(type: string): any {
```

Comment:
If we truly don't know what this returns, should this be `unknown` rather than `any`?